lib/src/models/mixins/voteable.dart

```diff
@@ -25,9 +32,35 @@ abstract class VoteableMixin implements RedditBaseInitializedMixin {
   /// The karma score of the voteable item.
   int get score => data['score'];
 
-  Future<void> _vote(String direction) async =>
-      reddit.post(apiPath['vote'], {'dir': direction, 'id': fullname},
-          discardResponse: true);
+  /// Has the currently authenticated [User] voted on this [VoteableMixin].
+  ///
+  /// returns [VoteState.upvoted] if has been upvoted,
```

Comment:
Change this comment to:

```dart
/// Has the currently authenticated [User] voted on this [UserContent].
///
/// Returns [VoteState.upvoted] if the content has been upvoted,
/// [VoteState.downvoted] if it has been downvoted, and [VoteState.none]
/// otherwise.
```